pkg/operator/nfs/controller.go

```diff
@@ -424,3 +429,37 @@ func (c *Controller) onDelete(obj interface{}) {
 	cluster := obj.(*nfsv1alpha1.NFSServer).DeepCopy()
 	logger.Infof("cluster %s deleted from namespace %s", cluster.Name, cluster.Namespace)
 }
+
+func validateNFSServerSpec(spec nfsv1alpha1.NFSServerSpec) error {
+	serverConfig := spec.Exports
+	for _, export := range serverConfig {
+		if err := validateAccessMode(export.Server.AccessMode); err != nil {
+			return err
+		}
+		if err := validateSquashMode(export.Server.Squash); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func validateAccessMode(mode string) error {
+	switch mode {
+	case "ReadOnly":
```

Comment:
does the underlying NFS server config allow different casing here? e.g., would `readonly` be OK?  if so, we need to normalize casing here for this check.

Reply:
OK so after getting the time to dive deeper into this, the [nfs-ganesha export config](https://github.com/nfs-ganesha/nfs-ganesha/blob/master/src/config_samples/config.txt) has squash options all in lowercase which I guess we will want to conform to. 
AccessMode/AccessType is slightly different, where nfs-ganesha lists e.g. `None`, `RW`, etc. This highlights maybe another necessity, as we would need to transform the CRD to the correct values for the ganesha export and this is already started in the `createGaneshaExport` func however this is only handling `ReadOnly` -> `RW` and not covering the other AccessTypes which will need to be done eventually... 

So kinda what I am thinking is for the validation step to ignore case for these values via my suggestion above, and handle transformations for `AccessMode` and `Squash` in `createGaneshaExport`. Does that works for everyone @jbw976 @rohan47? I am totally open to comments/suggestions here.